components/brave_rewards/resources/android_page/reducers/wallet_reducer.ts

```diff
@@ -0,0 +1,249 @@
+  chrome.send('brave_rewards.getRecurringTips')
+  chrome.send('brave_rewards.getOneTimeTips')
+  chrome.send('brave_rewards.getAdsData')
+
+  if (!state.safetyNetFailed) {
+    chrome.send('brave_rewards.getGrants', ['', ''])
+  }
+}
+
+const walletReducer: Reducer<Rewards.State | undefined> = (state: Rewards.State, action) => {
+  switch (action.type) {
+    case types.CREATE_WALLET:
+      chrome.send('brave_rewards.createWalletRequested', [])
+      break
+    case types.WALLET_CREATED:
+      state = { ...state }
+      state = createWallet(state)
+      fetchRewardsInfo(state)
+      chrome.send('brave_rewards.saveAdsSetting', ['adsEnabled', 'true'])
+      break
+    case types.WALLET_CREATE_FAILED:
+      state = { ...state }
+      state.walletCreateFailed = true
+      break
+    case types.GET_WALLET_PROPERTIES:
+      chrome.send('brave_rewards.getWalletProperties', [])
+      break
+    case types.ON_WALLET_PROPERTIES: {
+      state = { ...state }
+      let ui = state.ui
+
+      // TODO NZ check why enum can't be used inside Rewards namespace
+      if (action.payload.properties.status === 1) {
+        ui.walletServerProblem = true
+      } else {
+        // TODO NZ don't just assign directly
+        state.contributionMonthly = action.payload.properties.monthlyAmount
+        state.walletInfo = action.payload.properties.wallet
+        ui.walletServerProblem = false
+      }
+
+      state = {
+        ...state,
+        ui
+      }
+      break
+    }
+    case types.GET_WALLLET_PASSPHRASE:
+      chrome.send('brave_rewards.getWalletPassphrase', [])
+      break
+    case types.ON_WALLLET_PASSPHRASE:
+      const value = action.payload.pass
+      if (value && value.length > 0) {
+        state = { ...state }
+        state.recoveryKey = value
+      }
+      break
+    case types.RECOVER_WALLET:
+      if (!action.payload.key || action.payload.key.length === 0) {
+        let ui = state.ui
+        ui.walletRecoverySuccess = false
+
+        state = {
+          ...state,
+          ui
+        }
+
+        break
+      }
+
+      chrome.send('brave_rewards.recoverWallet', [action.payload.key])
+      break
+    case types.ON_RECOVER_WALLET_DATA: {
+      state = { ...state }
+      const result = action.payload.properties.result
+      const balance = action.payload.properties.balance
+      const grants = action.payload.properties.grants
+      let ui = state.ui
+      let walletInfo = state.walletInfo
+
+      // TODO NZ check why enum can't be used inside Rewards namespace
+      ui.walletRecoverySuccess = result === 0
+      if (result === 0) {
+        walletInfo.grants = grants || []
+        chrome.send('brave_rewards.getWalletPassphrase', [])
+        ui.emptyWallet = balance <= 0
+        ui.modalBackup = false
+      }
+
+      state = {
+        ...state,
+        ui,
+        walletInfo
+      }
+      break
+    }
+    case types.ON_ADDRESSES: {
+      if (!action.payload.addresses) {
```

Comment:
this is not needed as we don't have add funds

Reply:
Removed.